internal/server/authn/oidc.go

```diff
@@ -43,6 +45,31 @@ func NewOIDC(domain, clientID, clientSecret, redirectURL string) OIDC {
 	}
 }
 
+// Validate tests if an identity provider has valid attributes to support user login
+func (o *oidcImplementation) Validate() error {
+	ctx := context.Background()
+	conf, _, err := o.clientConfig(ctx)
+	if err != nil {
+		logging.S.Debugf("error validating oidc provider: %s", err)
+		return ErrInvalidProviderURL
+	}
+
+	_, err = conf.Exchange(ctx, "test-code") // 'test-code' is a placeholder for a valid authorization code, it will always fail
+	if err != nil {
```

Comment:
Is it expected to return valid if error is neither client ID or secret related? Are there other errors that could be returned? Might be a good idea to explicitly check invalid authorization code and return nil and error otherwise?

Reply:
The error message for an invalid auth code varies a lot by OIDC provider so I'm leaving this one on the permissive side for now